### PR TITLE
Handle IDelegateCreationOperation in CFG

### PR DESCRIPTION
**Pull request by mavasani**

### CODE REVIEW

File: src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs [hunk 1]
```diff
@@ -4998,6 +4998,12 @@ public override IOperation VisitDeclarationPattern(IDeclarationPatternOperation
                 operation.Syntax, operation.Type, operation.ConstantValue, IsImplicit(operation));
         }
 
+        public override IOperation VisitDelegateCreation(IDelegateCreationOperation operation, int? captureIdForResult)
```
[the remark this answers is not in the document]
Comment (mavasani) [reply src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs]: > When creating CFG graphs do we attempt to represent compiler optimizations? 

From my current understanding, we don't attempt to reflect compiler optimizations, and try to preserve the original tree shape unless it has flow semantics. @AlekseyTs to confirm.

File: src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IDelegateCreationExpression.cs [hunk 2]
```diff
@@ -1821,5 +1821,150 @@ void Main()
...
+        public void DelegateCreation_NoControlFlow()
+        {
+            string source = @"
+using System;
+
+class C
+{
+    void M(Action a1, Action a2, Action a3, Action a4)
+    /*<bind>*/{
+        a1 = () => { };
```
[the remark this answers is not in the document]
Comment (333fred) [reply src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IDelegateCreationExpression.cs]: We don't currently try to represent these optimizations.

**Review by 333fred — APPROVED** [review]

**Review by jaredpar — APPROVED** [review]

**Comment by mavasani** [comment]
@dotnet-bot retest windows_release_vs-integration_prtest please

**Comment by mavasani** [comment]
@dotnet-bot retest windows_debug_vs-integration_prtest please

**Comment by mavasani** [comment]
@dotnet-bot retest windows_release_vs-integration_prtest please

**Comment by mavasani** [comment]
@dotnet-bot retest windows_debug_vs-integration_prtest please

**Comment by mavasani** [comment]
Seems like unrelated failures in [windows_debug_vs-integration_prtest](https://ci.dot.net/job/dotnet_roslyn/job/features_dataflow/job/windows_debug_vs-integration_prtest/261/). All integration tests are failing. FYI @dotnet/roslyn-infrastructure

**Comment by mavasani** [comment]
@dotnet-bot retest windows_debug_vs-integration_prtest please

**Review by AlekseyTs — APPROVED** [review]
LGTM (iteration 2)

**Pull request merged**